packages/firestore/test/unit/local/bundle_cache.test.ts

```diff
@@ -0,0 +1,207 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import { IndexedDbPersistence } from '../../../src/local/indexeddb_persistence';
+import { filter, orderBy, path } from '../../util/helpers';
+import * as persistenceHelpers from './persistence_test_helpers';
+import { TestBundleCache } from './test_bundle_cache';
+import { SnapshotVersion } from '../../../src/core/snapshot_version';
+import { Timestamp } from '../../../src/api/timestamp';
+import { Query } from '../../../src/core/query';
+import { JSON_SERIALIZER } from './persistence_test_helpers';
+import { ResourcePath } from '../../../src/model/path';
+import { NamedQuery } from '../../../src/core/bundle';
+
+describe('MemoryBundleCache', () => {
+  let cache: TestBundleCache;
+
+  beforeEach(async () => {
+    cache = await persistenceHelpers
+      .testMemoryEagerPersistence()
+      .then(persistence => new TestBundleCache(persistence));
+  });
+
+  genericBundleCacheTests(() => cache);
+});
+
+describe('IndexedDbBundleCache', () => {
+  if (!IndexedDbPersistence.isAvailable()) {
+    console.warn('No IndexedDB. Skipping IndexedDbBundleCache tests.');
+    return;
+  }
+
+  let cache: TestBundleCache;
+  let persistence: IndexedDbPersistence;
+  beforeEach(async () => {
+    persistence = await persistenceHelpers.testIndexedDbPersistence({
+      synchronizeTabs: true
```

Comment:
I would omit this setting. It poses questions as to why you added it here, but doesn't (yet) at much benefit - other than remind me that I forgot to reply to your question about multi-tab sync...

Reply:
Done.